File: algoexpert/python/Validate_Subsequence.py
def isValidSubsequence(array, sequence):
    seqIndex = 0
    arrayIndex = 0
    while arrayIndex < len(array) and seqIndex < len(sequence):
        if array[arrayIndex] == sequence[seqIndex]:
            seqIndex += 1
        arrayIndex += 1
    return seqIndex == len(sequence)

File: algoexpert/java/Validate_Subsequence.java
import java.util.*;

class Program {
  public static boolean isValidSubsequence(List<Integer> array, List<Integer> sequence) {
    int seqIndex = 0;
    int arrIndex = 0;
    while(arrIndex < array.size() && seqIndex < sequence.size()){
      if(array.get(arrIndex).equals(sequence.get(seqIndex))) {
        seqIndex++;
      }
      arrIndex++;
    }
    return seqIndex == sequence.size();
  }
}